reference/orders.v2.oas2.yml

```diff
@@ -99,6 +100,14 @@ paths:
 
         After the update, the PUT request clears all discounts and promotions applied to the changed order line items. Since the order data syncs with other ERP systems, like Amazon or eBay, the updated order returns to the default setting, removing any applied discounts.
         
+        To update order fees, include the fee id in the request body along with all relevant fee fields. Fees not included will be deleted. Fees with an id will be updated, and fees without an id will be created as new.
+        
+        **Notes**
+        
+        * Sub-resources like products in the /v2/orders PUT request behave like PATCH, updating only the provided fields. Fees, however, follow standard PUT semantics and fees in the request body will fully replace existing ones. 
+        To retain an existing fee, include it in the body with its associated id. 
+        * The values for cost_ex_tax, cost_inc_tax and cost_tax in the fees payload should reflect the tax rate associated with the tax_class_id. For a 10% tax rate, the difference between cost_inc_tax and cost_ex_tax should be 10%. If no tax_class_id is provided, the store's default "tax class for fee" will apply. Incorrect data may lead to issues in downstream operations like refunds.
```

Comment:
> Incorrect data may lead to issues in downstream operations like refunds

should we do the validation for this logic ? Otherwise it may lead to uncertainty when using the API because there's no guarantee that the data provided by client is correct and does not lead to problem in downstream operations ?

Reply:
it's actually hard to do real validation because that would involve calling real tax providers to calculate the tax to se if it matches with the provided value (and we currently only support basic tax but not automatic tax providers in v2 api)

That's the reason we want to call it out very explicitly so it's users' own responsibility to ensure the data is correct.